README.md

```diff
@@ -1,2 +1,175 @@
+       Visit: [Ultrafast Grid configuration](https://applitools.com/docs/topics/sdk/vg-configuration.html)
+    7. #### Setting correct Viewport Size:
+        1. Incorrect Viewport size will cause the tests to be skipped with exception
+           as `EyesException: Unable to set Viewport Size`.
+        2. In that case, follow the below process to set Viewport Size Correctly:
+            - Navigate to Site: `https://whatismyviewport.com/` to get the viewport size.
+            - Set the Viewport size in `VIEWPORT_SIZE` in format: `Widthxheight` in `applitools.properties` file of your
+              automation framework repo
+
+3. All the Applitools Batch, Config and UFG Browser configuration related code is present in file `
+   src/main/java/com/znsio/api/ApplitoolsConfigurationManager.java``.
+    1. To add Batch Properties, navigate to above java class and add code in
+       method: `setBatchProperties(BatchInfo batchInfo)` as per given syntax and example:
+       ```
+       batchInfo.addProperty(<Key>, <Value>);
+       ```
+       example:
+       ```
+       batchInfo.addProperty("Agent Name", System.getenv("AGENT_NAME"));
+       ```
+    2. To add Test Properties, navigate to above Java class and add code in
+       method `setTestProperties(Configuration eyesConfig)`as per given syntax and example:
+        ```
+       eyesConfig.setPropertyName(<Value>);
+       ```
+       example:
+       ```
+       eyesConfig.setAppName(rpProperties.getProperty(Config.APP_NAME));
+       ```
+    3. **Dynamic properties:** For setting any additional property to either batch/test properties which is not already
+       configured in `ApplitoolsConfigurationManager` class, set that attribute key and value at either System Property
+       level or at Environment variable. Any key with prefix `AP_BATCH_` (for Batch Property) or `AP_TEST_` (for Test
+       property) set at System property or Environment variable level will be set as Applitools's Batch/Test property
+       respectively. The method which takes care of this configuration is `setPropertiesFromSystemVariables` defined
+       inside `ApplitoolsConfigurationManager` class.
+       For instance, if you're setting environment variable like `export AP_BATCH_Version=0.0.1`
+       and `export AP_TEST_Version=0.0.1`, then on Applitools dashboard, you'll see the corresponding properties
+       as `Version:0.0.1` for Batch/Test property respectively
+4. **Pipeline configuration:**
+    1. Set an environment variable `IS_VISUAL` to `true` for configuring visual validations for
+       pipeline executions. This property takes precedence over the `IS_VISUAL` property value of `config.properties` in
+       your automation framework repo.
+    2. The `ApplitoolsConfigurationManager` class takes care of setting the following batch/test properties in case the
+       test execution is happening on CI (Pipeline)
+       ```
+       BUILD_ID
+       AGENT_NAME
```

Comment:
AGENT_NAME is not used

Reply:
We're using it in Batch Properties. Let me know if you want me to remove it.